com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DeleteServicesStep.java

```diff
@@ -160,20 +162,24 @@ private void deleteService(CloudControllerClient client, String serviceName) {
 
     private void processException(Exception e, CloudServiceInstance serviceInstance, String serviceName) {
         if (e instanceof CloudOperationException) {
-            e = evaluateCloudOperationException((CloudOperationException) e, serviceName);
+            e = evaluateCloudOperationException((CloudOperationException) e, serviceName, serviceInstance.getService()
+                .getLabel());
             if (e == null) {
                 return;
             }
         }
         wrapAndThrowException(e, serviceInstance, serviceName);
     }
 
-    private CloudOperationException evaluateCloudOperationException(CloudOperationException e, String serviceName) {
+    private CloudOperationException evaluateCloudOperationException(CloudOperationException e, String serviceName, String label) {
         if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
-            getStepLogger().warn(e, Messages.COULD_NOT_DELETE_SERVICE, serviceName);
+            getStepLogger().warn(MessageFormat.format(Messages.COULD_NOT_DELETE_SERVICE, serviceName), e,
+                ExceptionMessageTailMapper.computeExceptionMessageTail(configuration, CloudComponents.ServiceBroker, label));
             return null;
         }
         if (e.getStatusCode() == HttpStatus.BAD_GATEWAY) {
+            getStepLogger().error(MessageFormat.format(Messages.COULD_NOT_DELETE_SERVICE, serviceName), e,

```

Comment:
The same here. We should avoid call getStepLogger().error(..) in steps. Instead, throw CloudServiceBrokerException with extended error message.